### PR TITLE
Update pytest to latest version

### PR DESCRIPTION
**Pull request by ashb**
The `tests/conftest.py` is a bit of a hack, but needed to support running tests against a non-editable install (which is the recommended way as per https://docs.pytest.org/en/latest/explanation/goodpractices.html?highlight=tests%20outside#choosing-a-test-layout-import-rules )

Closes https://github.com/p1c2u/openapi-core/pull/159

### CODE REVIEW

**Comment by p1c2u** [comment]
Hi @ashb thanks for the PR. I have few questions:

> a bit of a hack

How not tacky way would look like?

>  support running tests against a non-editable install 

I'm curious, why you need to run these tests?

> recommended way as per https://docs.pytest.org/en/latest/explanation/goodpractices.html?highlight=tests%20outside#choosing-a-test-layout-import-rules

Do you mean "Tests outside application code" section or can you point me to exact  paragraph?

Thanks in advance.

**Comment by ashb** [comment]
@p1c2u I haven't run in to this issue before with other pytest projects, so I'm not sure _why_ I've hit it here (perhaps I have always just run with `pip install -e . && pytest`)

So there are two non-hacky ways we could do this. (I define hacky as having to tweak `sys.path`)

- Update docs and CI to do `pip install -e .` before running tests
- Convert to the `src/` and `tests/` layout which seems to be what Pytest recommends.

I don't have a strong preference either way. Do you?

**Comment by uranusjr** [comment]
Another way to do this is to use [pytest-pythonpath](https://pypi.org/project/pytest-pythonpath/) and set

```ini
[tool:pytest]
pythonpath = .
```

in `setup.cfg`.

But always installing the package is still the better idea.

I think a lot of project these days don't have that issue since they tend to use nox or tox to run tests instead (they automatically reinstall the package into the test environment before invoking the test suite).

**Comment by p1c2u** [comment]
Thanks guys,

Python 2 support was dropped.  Nothing stopping us now to update to pytest 6.

> Convert to the src/ and tests/ layout which seems to be what Pytest recommends.

If so then we should go with the recommended way.

**Comment by ashb** [comment]
:+1: I'll update this to use the recommended way.

**Comment by ashb** [comment]
@p1c2u Updated this PR.

I've left it with the same source layout, but added `-e .` in to requirements_dev.txt` so that it's installed that way.

File: setup.cfg [hunk 1]
```diff
@@ -44,7 +44,7 @@ tests_require =
     webob
 
 [options.packages.find]
-exclude =
+exclude=
```
Comment (uranusjr) [inline setup.cfg:47]: Doesn’t seem to be needed?

File: setup.cfg [hunk 2]
```diff
@@ -34,7 +34,7 @@ install_requires =
     more-itertools
     dataclasses; python_version=="3.6"
 tests_require =
-    pytest>=5.0.0
+    pytest>=6.0.0
```
Comment (uranusjr) [inline setup.cfg:37]: Since we are always using pytest to run the tests, `tests_require` no longer used anywhere and can be removed.

Comment (ashb) [reply setup.cfg]: Thanks.

File: setup.cfg (hunk 1 above)
Comment (ashb) [reply setup.cfg:38]: Oh Correct. I failed to revert this space-only change as I went back-and-forth a few ways of doing things.

**Comment by p1c2u** [comment]
Updated pytest with new poetry build system

**Comment by p1c2u** [comment]
@ashb  @uranusjr thank you for the contribution

**Pull request merged**